### PR TITLE
chore(headless,atomic): add type "module" to atomic, atomic-react and headless

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-3580

## [Publint](https://publint.dev/) status
- atomic : no errors
- headless: no errors
- atomic-react: no errors

## [Are the types wrong](https://arethetypeswrong.github.io/) status
- atomic : errors Internal resolution in node16 (Do we need to address this ?) + cjs type warnings
- headless: errors Internal resolution in node16 ([addressed here](https://github.com/coveo/ui-kit/pull/4445))  + cjs type warnings
- atomic-react: cjs type warnings

### CODE REVIEW

File: packages/samples/vuejs/README.md [hunk 1]
```diff
@@ -101,12 +101,12 @@ scripts: {
...
 The Atomic package allows you to customize your components by defining CSS variables in your own stylesheet (see [Themes and Visual Customization](https://docs.coveo.com/en/atomic/latest/usage/themes-and-visual-customization/)).
 
-It also offers you a default theme that you can use as is or build upon. Those styles live in `@coveo/atomic/dist/atomic/themes/coveo.css` and you can import them in your entry js file as follows.
+It also offers you a default theme that you can use as is or build upon. Those styles live in `@coveo/atomic/themes/coveo.css` and you can import them in your entry js file as follows.
 
 ```ts
-import '@coveo/atomic/dist/atomic/themes/coveo.css';
 // optional!
 import {applyPolyfills, defineCustomElements} from '@coveo/atomic/loader';
+import '@coveo/atomic/themes/coveo.css';
```
Comment (louis-bompart) [inline packages/samples/vuejs/README.md:109]:
```suggestion
import {applyPolyfills, defineCustomElements} from '@coveo/atomic/loader';
// optional!
import '@coveo/atomic/themes/coveo.css';
```

File: packages/headless/package.json [hunk 2]
```diff
@@ -137,7 +138,7 @@
     "promote:npm:latest": "node ../../scripts/deploy/update-npm-tag.mjs latest",
     "build:doc": "npm run build:doc:extract && npm run build:doc:parse",
     "build:doc:extract": "node ./scripts/extract-documentation.mjs",
-    "build:doc:parse": "ts-node --project ./doc-parser/tsconfig.build.json ./doc-parser/doc-parser.ts"
+    "build:doc:parse": "exit 0"
```
Comment (louis-bompart) [inline packages/headless/package.json:141]: - [ ] Make ts-node works on ESM.

File: packages/atomic/package.json [hunk 3]
```diff
@@ -10,18 +11,36 @@
   },
   "main": "dist/index.cjs.js",
   "module": "dist/index.js",
-  "es2015": "dist/esm/index.mjs",
-  "es2017": "dist/esm/index.mjs",
   "types": "dist/types/index.d.ts",
-  "collection": "collection-manifest.json",
-  "collection:main": "dist/index.js",
-  "unpkg": "dist/atomic/atomic.esm.js",
```
Comment (louis-bompart) [inline packages/atomic/package.json:18]: Clean up non-standard.

File: packages/atomic/package.json [hunk 4]
```diff
@@ -10,18 +11,36 @@
...
+      "require": "./dist/atomic/assets/*"
+    },
+    "./lang/*": {
+      "import": "./dist/atomic/lang/*",
+      "require": "./dist/atomic/lang/*"
+    }
+  },
   "files": [
     "dist/",
     "loader/",
```
Comment (louis-bompart) [inline packages/atomic/package.json:41]:
```suggestion
```

File: packages/atomic/package.json [hunk 5]
```diff
@@ -66,7 +85,7 @@
   "devDependencies": {
     "@axe-core/playwright": "4.9.1",
     "@babel/core": "7.24.9",
-    "@coveo/atomic": "file:.",
```
Comment (louis-bompart) [inline packages/atomic/package.json:69]: We don't want to have the storybook utils in our "official" exports, so let's tweak this like so

File: packages/atomic/tsconfig.storybook.json [hunk 6]
```diff
@@ -4,7 +4,8 @@
     "emitDecoratorMetadata": true,
     "composite": true,
     "jsxFactory": "h",
-    "declaration": true
+    "declaration": true,
+    "allowImportingTsExtensions": true
```
Comment (louis-bompart) [inline packages/atomic/tsconfig.storybook.json:8]:
```suggestion
    "declaration": true
```

File: packages/headless/doc-parser/package.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "commonjs",
+  "dependencies": {
+    "@microsoft/api-extractor-model": "7.29.3",
+    "@microsoft/tsdoc": "0.15.0"
+  }
+}
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/headless/doc-parser/package.json:7]: Not really, its a sub to both make TS node interpret all this as CKS & to calm eslint

File: packages/atomic/package.json [hunk 8]
```diff
@@ -39,7 +39,8 @@
   "files": [
     "dist/",
     "docs/",
-    "licenses/"
+    "licenses/",
+    "loader/"
```
Comment (louis-bompart) [inline packages/atomic/package.json:43]: We should not loader anymore I think, wdyt @alexprudhomme ?

**Review by pathiery — APPROVED** [review]

**Review by alexprudhomme — APPROVED** [review]

File: patches/fix-esm-import-path+1.10.0.patch [hunk 9]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/fix-esm-import-path/fix-esm-import-path.js b/node_modules/fix-esm-import-path/fix-esm-import-path.js
```
Comment (fbeaudoincoveo) [inline patches/fix-esm-import-path+1.10.0.patch:1]: For the record, see https://github.com/beenotung/fix-esm-import-path/pull/14

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic-react/package.json [hunk 10]
```diff
@@ -11,6 +12,7 @@
   "scripts": {
     "build": "nx build",
     "clean": "rimraf -rf dist",
+    "build:addExtensions": "fix-esm-import-path src/components/stencil-generated",
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic-react/package.json:15]: - [ ] better script name

File: packages/atomic/.storybook/main.mts [hunk 11]
```diff
@@ -8,7 +8,10 @@ const isCDN = process.env.DEPLOYMENT_ENVIRONMENT === 'CDN';
 
 const config: StorybookConfig = {
   stories: ['../src/**/*.new.stories.@(js|jsx|ts|tsx|mdx)'],
-  staticDirs: [{from: '../dist/atomic', to: './assets'}],
+  staticDirs: [
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/.storybook/main.mts:11]: Not required, because it is imported in the preview.ts, see
packages/atomic/.storybook/preview.ts#L1

**Comment by alexprudhomme** [comment]
This PR was split in 4 sections

1. This PR
2. https://github.com/coveo/ui-kit/pull/4445
3. https://github.com/coveo/ui-kit/pull/4449
4. https://github.com/coveo/ui-kit/pull/4452

**Pull request merged**